src/probnum/quad/bq_methods/bq_state.py

```diff
@@ -0,0 +1,123 @@
+"""State of a Bayesian quadrature method."""
+
+from typing import Callable, Optional, Tuple
+
+import numpy as np
+
+from probnum.kernels._kernel import Kernel
+from probnum.quad._integration_measures import IntegrationMeasure
+from probnum.quad.kernel_embeddings import KernelEmbedding
+from probnum.randvars import Normal
+
+
+class BQInfo:
+    """Collect information about the BQ loop.
+
+    Parameters
+    ----------
+    iteration :
+        Iteration of the loop.
+    nevals :
+        Number of evaluations collected.
+    has_converged :
+        True if the BQ loop fulfils a stopping criterion, otherwise False.
+    stopping_criterion:
+        The stopping criterion used to determine convergence.
+    """
+
+    def __init__(
+        self,
+        iteration: int = 0,
+        nevals: int = 0,
+        has_converged: bool = False,
+        stopping_criterion=None,
+    ):
+        self.iteration = iteration
+        self.nevals = nevals
+        self.has_converged = has_converged
+        self.stopping_criterion = stopping_criterion
+
+    def update_iteration(self, batch_size):
+        """
+        Update the quantities tracking iteration info.
+        Parameters
+        ----------
+        batch_size:
+            Number of points added in each iteration.
+        """
+        self.iteration += 1
+        self.nevals += batch_size
+
+
+class BQState:
+    """Container for the quantities defining the BQ problem and the BQ loop state.
+
+    Parameters
+    ----------
+    measure :
+        The integration measure.
+    kernel :
+        The kernel used for BQ.
+    integral_belief :
+        Normal distribution over the value of the integral.
+    info:
+        Information about the loop status.
+    batch_size:
+        Size of the batch when acquiring new nodes.
+    nodes:
+        All locations at which function evaluations are available.
+    fun_evals:
+        Function evaluations at nodes.
+    """
+
+    def __init__(
+        self,
+        measure: IntegrationMeasure,
+        kernel: Kernel,
+        integral_belief: Optional[Normal] = None,
+        previous_integral_beliefs: Optional[Tuple[Normal]] = (),
```

Comment:
Because you might be interested in convergence behavior as a function of samples. Also there's a stopping criterion `RelativeError` that stops when the difference of consecutive integral estimates drop below a certain value.

Reply:
OK, thanks, this wasn;t obvious to me.